src/pts/pyspark/common/cast_to_schema.py

```diff
@@ -0,0 +1,145 @@
+                target_field.dataType,
+            ),
+        )
+
+    return df
+
+
+def parse_spark_schema(schema_json: str) -> StructType:
+    """Parse Spark schema from JSON.
+
+    Args:
+        schema_json (str): JSON filename containing spark schema in the schemas package
+
+    Returns:
+        StructType: Spark schema
+    """
+    pkg = pkg_resources.files(schemas).joinpath(schema_json)
+    with pkg.open(encoding='utf-8') as schema:
+        core_schema = json.load(schema)
+
+    return StructType.fromJson(core_schema)
+
+
+def cast_column_to_target_type(
+    col: Column,
+    source_type: DataType,
+    target_type: DataType,
+) -> Column:
+    """Cast a column to match the target type.
+
+    Handles arrays, and nested structs recursively.
+
+    Args:
+        col: The column expression to cast
+        source_type: The source data type
+        target_type: The target data type
+
+    Returns:
+        The cast column expression
+    """
+    if isinstance(source_type, ArrayType) and isinstance(target_type, ArrayType):
+        if isinstance(target_type.elementType, StructType):
+            # Array of structs
+            return f.transform(
+                col,
+                lambda x: cast_struct(x, source_type.elementType, target_type.elementType),
+            )
+        else:
+            # Simple array
+            return f.transform(col, lambda x: x.cast(target_type.elementType))
+
+    if isinstance(source_type, StructType) and isinstance(target_type, StructType):
+        return cast_struct(col, source_type, target_type)
+
+    # Simple casts
+    return col.cast(target_type)
+
+
+def cast_struct(
+    struct_col: Column,
+    source_struct_type: StructType,
+    target_struct_type: StructType,
+) -> Column:
+    """Cast a struct field to match the target struct schema.
+
+    Handles missing fields, type conversions, and field reordering.
+
+    Args:
+        struct_col: The struct column to cast
+        source_struct_type: The source struct type
+        target_struct_type: The target struct type
+
+    Returns:
+        A struct with fields matching the target schema
+    """
+    source_fields = {field.name: field for field in source_struct_type.fields}
+
+    field_exprs = []
+    for target_field in target_struct_type.fields:
+        field_name = target_field.name
+
+        if field_name in source_fields:
+            # Field exists - get it and cast to target type
+            field_expr = struct_col.getField(field_name)
+            source_field_type = source_fields[field_name].dataType
+
+            # Recursively handle nested types
+            if isinstance(target_field.dataType, (StructType, ArrayType)):
+                field_expr = cast_column_to_target_type(
+                    field_expr,
+                    source_field_type,
+                    target_field.dataType,
+                )
+            else:
+                field_expr = field_expr.cast(target_field.dataType)
```

Comment:
I don't think this condition is required here. The `cast_column_to_target_type` method can deal with all complex and simple types.

Reply:
Phew, great catch. You're right, this is redundant. Addressed